pkg/machinery/resources/network/network.go

```diff
@@ -36,17 +36,21 @@ func LinkID(linkName string) string {
 }
 
 // RouteID builds ID (primary key) for the route.
-func RouteID(table nethelpers.RoutingTable, family nethelpers.Family, destination netip.Prefix, gateway netip.Addr, priority uint32) string {
+func RouteID(table nethelpers.RoutingTable, family nethelpers.Family, destination netip.Prefix, gateway netip.Addr, priority uint32, outLinkName string) string {
 	dst, _ := destination.MarshalText() //nolint:errcheck
 	gw, _ := gateway.MarshalText()      //nolint:errcheck
 
-	tablePrefix := ""
+	prefix := ""
 
 	if table != nethelpers.TableMain {
-		tablePrefix = fmt.Sprintf("%s/", table)
+		prefix = fmt.Sprintf("%s/", table)
 	}
 
-	return fmt.Sprintf("%s%s/%s/%s/%d", tablePrefix, family, string(gw), string(dst), priority)
+	if family == nethelpers.FamilyInet6 {
+		prefix += fmt.Sprintf("%s/", outLinkName)
+	}
```

Comment:
is the issue only with ipv6? cannot the same problem happen with ipv4?

Reply:
nope, kernel doesn't allow that (that's why I made the route ID this way initially), but IPv6 is special :)